# Week7 - Decision Tree Lab

* Train-test split
* Train a decison tree model
* Train a random forest model
* Evaluate the models
* Explain findings

In [134]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from sklearn.tree import DecisionTreeClassifier
%matplotlib inline

df = pd.read_csv('https://raw.githubusercontent.com/msaricaumbc/DS_data/master/ds602/log_reg/employee-turnover-balanced.csv')
y = df['left_company']
X = df.iloc[:, 1:]

# Train-test split

In [144]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)

# Train Decision Tree model and Random Forest model

In [145]:
from sklearn.tree import DecisionTreeClassifier
# train decision tree classifier
dt = DecisionTreeClassifier(random_state=42)
dt.fit(X_train, y_train)

DecisionTreeClassifier(random_state=42)

In [146]:
from sklearn.ensemble import RandomForestClassifier
# train random forest classifier
rf = RandomForestClassifier(random_state=42)
rf.fit(X_train, y_train)


RandomForestClassifier(random_state=42)

# Evaluate the decision model and random forest model

In [147]:
from sklearn.metrics import accuracy_score, classification_report

# evaluate decision tree model
y_pred_decision = dt.predict(X_test)
acc_dtc = accuracy_score(y_test, y_pred_decision)
print("Decision Tree model Accuracy:", acc_dtc)
print(classification_report(y_test, y_pred_decision))

# evaluate random forest model
y_pred_random = rf.predict(X_test)
acc_rfc = accuracy_score(y_test, y_pred_random)
print("Random Forest model Accuracy:", acc_rfc)
print(classification_report(y_test, y_pred_random))

Decision Tree model Accuracy: 0.7966666666666666
              precision    recall  f1-score   support

          No       0.84      0.73      0.78       150
         Yes       0.76      0.87      0.81       150

    accuracy                           0.80       300
   macro avg       0.80      0.80      0.80       300
weighted avg       0.80      0.80      0.80       300

Random Forest model Accuracy: 0.87
              precision    recall  f1-score   support

          No       0.89      0.85      0.87       150
         Yes       0.85      0.89      0.87       150

    accuracy                           0.87       300
   macro avg       0.87      0.87      0.87       300
weighted avg       0.87      0.87      0.87       300



# Explain Findings


As can be seen from the results above, the decision tree model's accuracy is 0.796 while the random forest model's accuracy is 0.87. We may therefore conclude that the Random Forest model is superior to the Decision Tree model by comparing their accuracy scores.